File: lab3/src/main/java/lab3/myquest/locations/Space.java
package lab3.myquest.locations;

import lab3.functionality.Action;
import lab3.functionality.Item;
import lab3.functionality.Location;
import lab3.functionality.Player;

public class Space extends Location {
    public Space(String name) {
        super(name, p -> p.getName() + " понимает, что парит над полом!");
        var shell = new Item("Шкаф", "Шкаф для книг с открытой дверцей. Что он здесь делает?");
        shell.addAction(new Action("Закрыть дверцу", p -> p.getName() + " закрыл дверцу шкафа", (q, p) -> shell.setDescription("Шкаф для книг с закрытой дверцей."), true));
        addAction(new Action("Прислушаться к ощущениям", p -> String.format("Вместе с головокружением у %s появилось какое-то странное ощущение зависания вниз головой, то есть ему на какой-то миг показалось, будто он перевернут кверху ногами", p.getName()), (q, p) -> {
                    System.out.println(p.getName() + " убедился, что он вовсе не вверх ногами.");
                    addItem(shell);
                }, true)
        );
    }

    @Override
    public String enter(Player player) {
        return "";
    }
}

File: lab4/src/main/kotlin/lab4/myQuest/locations/Space.kt
package lab4.myQuest.locations

import lab4.functionality.ActionFactory
import lab4.functionality.Location
import lab4.functionality.Player
import lab4.myQuest.items.Shell

class Space : Location("Пространство", { it.name + " понимает, что парит над полом!" }) {
    init {
        actions.add(
            ActionFactory().addItem(
                "Прислушаться к ощущениям",
                { "Вместе с головокружением у ${it.name} появилось какое-то странное ощущение зависания вниз головой, то есть ему на какой-то миг показалось, будто он перевернут кверху ногами\n${it.name} убедился, что он вовсе не вверх ногами." },
                Shell()
            )
        )
        actions.add(
            ActionFactory().changeLocation(
                "Выпрямиться",
                { "${it.name} хотел выпрямиться, но как раз в это время его снизу словно толкнуло что-то и подбросило под потолок.\n${it.name} ударился головой об потолок." },
                Ceil()
            )
        )
    }

    override fun enter(player: Player): String {
        return ""
    }
}